### PR TITLE
[LM-9407] Added Tagging

### PR DESCRIPTION
**Pull request by prabha-likeminds**
## Description

**Features**
- Added Tagging functionality

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 1]
```diff
@@ -169,6 +173,16 @@ const UniversalFeed = () => {
...
   const postAdd = useCallback(async () => {
+    // replace the mentions with route
+    const postContentText = replaceMentionValues(postContent, ({id, name}) => {
+      const PATH = extractPathfromRouteQuery(id);
+      if (!PATH) {
+        return `<<${name}|route://${name}>>`;
+      } else {
+        return `<<${name}|route://${id}>>`;
+      }
+    });
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:184]: move this function to a util function

File: FeedSX/screens/createPost/index.tsx [hunk 2]
```diff
@@ -102,6 +109,14 @@ const CreatePost = (props: IProps) => {
   const postToEdit = props?.route?.params;
   const [postDetail, setPostDetail] = useState({} as LMPostUI);
   const [postContentText, setPostContentText] = useState('');
+  const myRef = useRef<any>();
+  const [taggedUserName, setTaggedUserName] = useState('');
+  const [debounceTimeout, setDebounceTimeout] = useState<any>(null);
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:114]: why `any`?

File: FeedSX/screens/createPost/index.tsx [hunk 3]
```diff
@@ -102,6 +109,14 @@ const CreatePost = (props: IProps) => {
   const postToEdit = props?.route?.params;
   const [postDetail, setPostDetail] = useState({} as LMPostUI);
   const [postContentText, setPostContentText] = useState('');
+  const myRef = useRef<any>();
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:112]: why `any`?

File: FeedSX/screens/createPost/index.tsx [hunk 4]
```diff
@@ -102,6 +109,14 @@ const CreatePost = (props: IProps) => {
...
+  const [taggedUserName, setTaggedUserName] = useState('');
+  const [debounceTimeout, setDebounceTimeout] = useState<any>(null);
+  const [page, setPage] = useState(1);
+  const [userTaggingListHeight, setUserTaggingListHeight] = useState<any>(116);
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:116]: why `any`?

File: FeedSX/screens/createPost/index.tsx [hunk 5]
```diff
@@ -102,6 +109,14 @@ const CreatePost = (props: IProps) => {
...
+  const [allTags, setAllTags] = useState<any>([]);
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:117]: why `any`?

File: FeedSX/screens/createPost/index.tsx [hunk 6]
```diff
@@ -348,7 +363,18 @@ const CreatePost = (props: IProps) => {
...
+        ({URLwithID, name}) => {
+          if (!URLwithID) {
+            return `@[${name}](${name})`;
+          } else {
+            return `@[${name}](${URLwithID})`;
+          }
+        },
+      );
+      setPostContentText(convertedText);
+      // setPostContentText(postDetail?.text);
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:377]: remove commented code

File: FeedSX/screens/createPost/index.tsx [hunk 7]
```diff
@@ -348,7 +363,18 @@ const CreatePost = (props: IProps) => {
...
+      const convertedText = convertToMentionValues(
+        `${postDetail?.text} `, // to put extra space after a message whwn we want to edit a message
+        ({URLwithID, name}) => {
+          if (!URLwithID) {
+            return `@[${name}](${name})`;
+          } else {
+            return `@[${name}](${URLwithID})`;
+          }
+        },
+      );
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:375]: move this function to a util

File: FeedSX/screens/createPost/index.tsx [hunk 8]
```diff
@@ -373,13 +399,24 @@ const CreatePost = (props: IProps) => {
...
+    // replace mentions with route
+    const contentText = replaceMentionValues(postContentText, ({id, name}) => {
+      // example ID = `user_profile/8619d45e-9c4c-4730-af8e-4099fe3dcc4b`
+      const PATH = extractPathfromRouteQuery(id);
+      if (!PATH) {
+        return `<<${name}|route://${name}>>`;
+      } else {
+        return `<<${name}|route://${id}>>`;
+      }
+    });
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:411]: move this to a util class

File: FeedSX/screens/createPost/index.tsx [hunk 9]
```diff
@@ -389,6 +426,88 @@ const CreatePost = (props: IProps) => {
     return editPostResponse;
   };
 
+  // this function is called on change text of inputText
+  const handleInputChange = async (event: any) => {
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:430]: why `any`?

File: FeedSX/screens/createPost/index.tsx [hunk 10]
```diff
@@ -416,13 +535,82 @@ const CreatePost = (props: IProps) => {
...
+          <View
+            style={[
+              styles.taggingListView,
+              {
+                height: userTaggingListHeight,
+              },
+            ]}>
+            <FlashList
+              data={[...allTags]}
+              renderItem={({item}: any) => {
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:563]: why `any`?

Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:563]: rename `item` to `user`

File: FeedSX/screens/createPost/index.tsx [hunk 11]
```diff
@@ -416,13 +535,82 @@ const CreatePost = (props: IProps) => {
...
+                      setIsUserTagging(false);
+                    }}
+                    style={styles.taggingListItem}>
+                    <LMProfilePicture
+                      fallbackText={item?.name}
+                      fallbackTextBoxStyle={styles.taggingListProfileBoxStyle}
+                      size={40}
+                    />
+                    <View style={styles.taggingListItemTextView}>
+                      <Text style={styles.taggingListText} numberOfLines={1}>
```
Comment (ishaan1607) [inline FeedSX/screens/createPost/index.tsx:585]: we should use `LMText`

File: FeedSX/screens/postDetail/index.tsx [hunk 12]
```diff
@@ -119,6 +136,14 @@ const PostDetail = (props: IProps) => {
     useState(showDeleteModal);
   const [keyboardIsVisible, setKeyboardIsVisible] = useState(false);
   const [editCommentFocus, setEditCommentFocus] = useState(false);
+  const myRef = useRef<any>();
+  const [taggedUserName, setTaggedUserName] = useState('');
+  const [debounceTimeout, setDebounceTimeout] = useState<any>(null);
```
Comment (ishaan1607) [inline FeedSX/screens/postDetail/index.tsx:141]: why `any`?

File: FeedSX/screens/postDetail/index.tsx [hunk 13]
```diff
@@ -119,6 +136,14 @@ const PostDetail = (props: IProps) => {
...
+  const [page, setPage] = useState(1);
+  const [userTaggingListHeight, setUserTaggingListHeight] = useState<any>(116);
```
Comment (ishaan1607) [inline FeedSX/screens/postDetail/index.tsx:143]: why `any`?

File: FeedSX/screens/postDetail/index.tsx [hunk 14]
```diff
@@ -119,6 +136,14 @@ const PostDetail = (props: IProps) => {
...
+  const [allTags, setAllTags] = useState<any>([]);
```
Comment (ishaan1607) [inline FeedSX/screens/postDetail/index.tsx:144]: why `any`?

File: FeedSX/screens/postDetail/index.tsx [hunk 15]
```diff
@@ -262,7 +290,18 @@ const PostDetail = (props: IProps) => {
...
+      // converts the mentions route to mention values
+      const convertedComment = convertToMentionValues(
+        `${commentDetail?.text} `, // to put extra space after a message whwn we want to edit a message
+        ({URLwithID, name}) => {
+          if (!URLwithID) {
+            return `@[${name}](${name})`;
+          } else {
+            return `@[${name}](${URLwithID})`;
+          }
+        },
```
Comment (ishaan1607) [inline FeedSX/screens/postDetail/index.tsx:302]: move this function to a util

File: FeedSX/screens/postDetail/index.tsx [hunk 16]
```diff
@@ -354,14 +393,28 @@ const PostDetail = (props: IProps) => {
...
+    const convertedNewComment = replaceMentionValues(
+      commentToAdd,
+      ({id, name}) => {
+        const PATH = extractPathfromRouteQuery(id);
+        if (!PATH) {
+          return `<<${name}|route://${name}>>`;
+        } else {
+          return `<<${name}|route://${id}>>`;
+        }
+      },
```
Comment (ishaan1607) [inline FeedSX/screens/postDetail/index.tsx:406]: move this function to a util

File: FeedSX/screens/postDetail/index.tsx [hunk 17]
```diff
@@ -376,15 +429,28 @@ const PostDetail = (props: IProps) => {
...
+    const convertedNewReply = replaceMentionValues(
+      commentToAdd,
+      ({id, name}) => {
+        const PATH = extractPathfromRouteQuery(id);
+        if (!PATH) {
+          return `<<${name}|route://${name}>>`;
+        } else {
+          return `<<${name}|route://${id}>>`;
+        }
+      },
```
Comment (ishaan1607) [inline FeedSX/screens/postDetail/index.tsx:442]: move this function to a util, as this duplicate function called in various places

File: FeedSX/screens/postDetail/index.tsx [hunk 18]
```diff
@@ -494,17 +560,30 @@ const PostDetail = (props: IProps) => {
...
+    const convertedEditedComment = replaceMentionValues(
+      commentToAdd,
+      ({id, name}) => {
+        const PATH = extractPathfromRouteQuery(id);
+        if (!PATH) {
+          return `<<${name}|route://${name}>>`;
+        } else {
+          return `<<${name}|route://${id}>>`;
+        }
+      },
```
Comment (ishaan1607) [inline FeedSX/screens/postDetail/index.tsx:573]: move this function to a util, as this duplicate function called in various places

File: FeedSX/screens/postDetail/index.tsx [hunk 19]
```diff
@@ -703,8 +861,76 @@ const PostDetail = (props: IProps) => {
...
+              {
+                paddingBottom: replyOnComment.textInputFocus
+                  ? Layout.normalize(74)
+                  : Layout.normalize(44),
+                height: userTaggingListHeight,
+              },
+            ]}>
+            <FlashList
+              data={[...allTags]}
+              renderItem={({item}: any) => {
```
Comment (ishaan1607) [inline FeedSX/screens/postDetail/index.tsx:878]: why `any`?

Comment (ishaan1607) [inline FeedSX/screens/postDetail/index.tsx:878]: rename `item` to `user`

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: FeedSX/screens/postDetail/index.tsx (hunk 19 above)
Comment (prabha-likeminds) [reply FeedSX/screens/postDetail/index.tsx:878]: item is used as standard name in flashlist

File: FeedSX/screens/createPost/index.tsx (hunk 10 above)
Comment (prabha-likeminds) [reply FeedSX/screens/createPost/index.tsx:563]: item is used as standard name in flashlist

**Review by ishaan1607 — APPROVED** [review]

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**